tests/conftest.py

```diff
@@ -1,138 +1,391 @@
+        client_id="CLIENT_ID",
+        client_secret=str(_secret_path),
+        client_machine="MACHINE_NAME",
+        small_file_size_mb=1.0,
+        temp_directory=str(_temp_directory),
+    )
+    computer.store()
+    return computer
+
+
+class MockFirecrest:
+    def __init__(self, firecrest_url, *args, **kwargs):
+        self._firecrest_url = firecrest_url
+        self.args = args
+        self.kwargs = kwargs
+
+        self.whoami = whomai
+        self.list_files = list_files
+        self.stat = stat_
+        self.mkdir = mkdir
+        self.simple_delete = simple_delete
+        self.parameters = parameters
+        self.symlink = symlink
+        self.checksum = checksum
+        self.simple_download = simple_download
+        self.simple_upload = simple_upload
+        self.compress = compress
+        self.extract = extract
+        self.copy = copy
+        self.submit = submit
+        self.poll_active = poll_active
+
+
+class MockClientCredentialsAuth:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
 
 
 @pytest.fixture(scope="function")
-def firecrest_server(
+def myfirecrest(
     pytestconfig: pytest.Config,
-    request: pytest.FixtureRequest,
     monkeypatch,
-    tmp_path: Path,
 ):
-    """A fixture which provides a mock Firecrest server to test against."""
-    config_path: str | None = request.config.getoption("--firecrest-config")
-    no_clean: bool = request.config.getoption("--firecrest-no-clean")
-    record_requests: bool = request.config.getoption("--firecrest-requests")
-    telemetry: RequestTelemetry | None = None
-
-    if config_path is not None:
-        # if given, use this config
-        with open(config_path, encoding="utf8") as handle:
-            config = json_load(handle)
-        config = FirecrestConfig(**config)
-        # rather than use the scratch_path directly, we use a subfolder,
-        # which we can then clean
-        config.scratch_path = config.scratch_path + "/pytest_tmp"
-
-        # we need to connect to the client here,
-        # to ensure that the scratch path exists and is empty
-        client = f7t.Firecrest(
-            firecrest_url=config.url,
-            authorization=f7t.ClientCredentialsAuth(
-                config.client_id, config.client_secret, config.token_uri
-            ),
+    monkeypatch.setattr(firecrest, "Firecrest", MockFirecrest)
+    monkeypatch.setattr(firecrest, "ClientCredentialsAuth", MockClientCredentialsAuth)
+
+
+def submit(
+    machine: str,
+    script_str: Optional[str] = None,
+    script_remote_path: Optional[str] = None,
+    script_local_path: Optional[str] = None,
+    local_file=False,
+):
+    if local_file:
+        raise DeprecationWarning("local_file is not supported")
+
+    if script_remote_path and not Path(script_remote_path).exists():
+        raise FileNotFoundError(f"File {script_remote_path} does not exist")
+    job_id = random.randint(10000, 99999)
```

Comment:
Not exactly sure what you mean @khsrali. Because this is for mocking, the job_id does not matter here? Can't you hardcode it if the number does not matter? Using random without a seed is a pattern that looks like opening the door for surprising behaviors or making debugging harder.